### PR TITLE
Adding Context Creation Test for OffscreenCanvas

### PR DESCRIPTION
**Pull request by Juanmihd**
This PR adds the test needed to verify that an OffscreenCanvas can create a context of type WebGPU through the 'gpupresent' value.

This functionality is under review in https://chromium-review.googlesource.com/c/chromium/src/+/2720944

-----

- [x] New helpers, if any, are documented in `helper_index.md`.
- [X] Incomplete tests, if any, are marked with TODO or `.unimplemented()`.

**[Review requirement](https://github.com/gpuweb/cts/blob/main/docs/reviews.md) checklist:**

- [x] WebGPU readability
- [x] TypeScript readability

### CODE REVIEW

File: src/webgpu/web_platform/offscreencanvas/context_creation.spec.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    if (typeof document === 'undefined') {
+      // Skip if there is no document (Workers, Node)
+      t.skip('DOM is not available to create canvas element');
+    }
+
+    const canvas = new OffscreenCanvas(10,10);
+
+    // TODO: fix types so these aren't necessary
+    /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+    const ctx: any = canvas.getContext('gpupresent');
```
Comment (Kangz) [inline src/webgpu/web_platform/offscreencanvas/context_creation.spec.ts:24]: I think the explicit any is missing on this line.

**Review by Kangz — APPROVED** [review]
LGTM

**Review by kainino0x — APPROVED** [review]
LGTM. I realized some of the type mess can be cleaned up now so I'll just do that in this PR.

I also noticed we can avoid having duplication between these two tests by parameterizing them (easier with the types fix). @Juanmihd Would you review my additional changes?

**Review by Juanmihd — COMMENTED** [review]
LGTM, thanks for making the test better!

**Pull request merged**